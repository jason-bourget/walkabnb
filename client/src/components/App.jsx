import React from 'react';
import axios from 'axios';
import Listings from './Listings.jsx';
import Title from './Title.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listings: []
    }
  }

  getListings = async (city) => {
    const { data } = await axios.get('/api/listings', { params: { city } });
    this.setState({ listings: data });
  };

  render() {
    return (
      <div>
        <Title getListings={this.getListings}/>
        <Listings listings={this.state.listings}/>
      </div>
    )
  };
}

export default App;