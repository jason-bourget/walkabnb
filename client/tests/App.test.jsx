import React from 'react';
import App from '../src/components/App.jsx';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const component = renderer.create(
    <App/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});