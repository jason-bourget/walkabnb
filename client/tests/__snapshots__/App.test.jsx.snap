// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="Title-root-1"
  >
    <header
      className="MuiPaper-root-15 MuiPaper-elevation4-21 MuiAppBar-root-6 MuiAppBar-positionFixed-7 MuiAppBar-colorPrimary-13 mui-fixed"
      style={
        Object {
          "backgroundColor": "#0d47a1",
          "position": "fixed",
        }
      }
    >
      <div
        className="MuiToolbar-root-42 MuiToolbar-regular-44 MuiToolbar-gutters-43"
      >
        <h6
          className="MuiTypography-root-46 MuiTypography-h6-63 MuiTypography-colorInherit-75 MuiTypography-noWrap-72 Title-title-3"
        >
          walkabnb
        </h6>
        <div
          className="AutoSuggest-root-82"
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="AutoSuggest-container-83"
            role="combobox"
          >
            <div
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              className="MuiFormControl-root-88 MuiFormControl-fullWidth-91"
              onKeyDown={[Function]}
              style={Object {}}
            >
              <div
                className="MuiInputBase-root-105 MuiOutlinedInput-root-92 MuiInputBase-fullWidth-114 MuiInputBase-formControl-106"
                onClick={[Function]}
                style={
                  Object {
                    "borderTopColor": "white",
                    "color": "white",
                  }
                }
              >
                <fieldset
                  aria-hidden={true}
                  className="MuiPrivateNotchedOutline-root-122 MuiOutlinedInput-notchedOutline-99"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="MuiPrivateNotchedOutline-legend-123"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
                <input
                  aria-invalid={false}
                  autoComplete="off"
                  className="MuiInputBase-input-115 MuiOutlinedInput-input-100"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search for a city..."
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="MuiPaper-root-15 MuiPaper-elevation2-19"
              id="react-autowhatever-1"
              role="listbox"
              style={Object {}}
            />
          </div>
          <button
            className="MuiButtonBase-root-150 MuiButton-root-124 MuiButton-contained-135 MuiButton-raised-138 AutoSuggest-button-87"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-125"
            >
              Search
            </span>
            <span
              className="MuiTouchRipple-root-153"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <img
    alt="logo"
    height="200px"
    src=""
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "position": "relative",
        "top": "100px",
      }
    }
  />
</div>
`;
